pkg/manager/impl/execution_manager.go

```diff
@@ -140,40 +144,23 @@ func validateMapSize(maxEntries int, candidate map[string]string, candidateName
 	return nil
 }
 
-// Labels and annotations defined in the execution spec are preferred over those defined in the
-// reference launch plan spec.
-func (m *ExecutionManager) addLabelsAndAnnotations(requestSpec *admin.ExecutionSpec,
-	partiallyPopulatedInputs *workflowengineInterfaces.ExecuteWorkflowInput) error {
-
-	var labels map[string]string
-	if requestSpec.Labels != nil && requestSpec.Labels.Values != nil {
-		labels = requestSpec.Labels.Values
-	} else if partiallyPopulatedInputs.Reference.Spec.Labels != nil &&
-		partiallyPopulatedInputs.Reference.Spec.Labels.Values != nil {
-		labels = partiallyPopulatedInputs.Reference.Spec.Labels.Values
-	}
+type mapLikeThing interface {
+	GetValues() map[string]string
+}
 
-	var annotations map[string]string
-	if requestSpec.Annotations != nil && requestSpec.Annotations.Values != nil {
-		annotations = requestSpec.Annotations.Values
-	} else if partiallyPopulatedInputs.Reference.Spec.Annotations != nil &&
-		partiallyPopulatedInputs.Reference.Spec.Annotations.Values != nil {
-		annotations = partiallyPopulatedInputs.Reference.Spec.Annotations.Values
+func (m *ExecutionManager) resolveStringMap(preferredValues, defaultValues mapLikeThing, valueName string, maxEntries int) (map[string]string, error) {
```

Comment:
Should this  now be member function of ExecutionManager.

Earlier since it was using m.config.RegistrationValidationConfiguration().GetMaxLabelEntries() , but this seems to be standalone function now.

Reply:
good point, done